src/components/bottom_bar_statistic/aggregate_statistics_store.ts

```diff
@@ -0,0 +1,130 @@
+import { sum } from "../../functions/helper_math";
+import { average, countAny, countNumbers, max, min } from "../../functions/helper_statistical";
+import { positions } from "../../helpers";
+import { Get } from "../../store_engine";
+import { SpreadsheetStore } from "../../stores";
+import { _t } from "../../translation";
+import { CellValueType, Command, EvaluatedCell, Locale, isCoreCommand } from "../../types";
+
+export interface StatisticFnResults {
+  [name: string]: number | undefined;
+}
+
+interface SelectionStatisticFunction {
+  name: string;
+  compute: (data: EvaluatedCell[], locale: Locale) => number;
+  types: CellValueType[];
+}
+
+const selectionStatisticFunctions: SelectionStatisticFunction[] = [
+  {
+    name: _t("Sum"),
+    types: [CellValueType.number],
+    compute: (values, locale) => sum([[values]], locale),
+  },
+  {
+    name: _t("Avg"),
+    types: [CellValueType.number],
+    compute: (values, locale) => average([[values]], locale),
+  },
+  {
+    name: _t("Min"),
+    types: [CellValueType.number],
+    compute: (values, locale) => min([[values]], locale),
+  },
+  {
+    name: _t("Max"),
+    types: [CellValueType.number],
+    compute: (values, locale) => max([[values]], locale),
+  },
+  {
+    name: _t("Count"),
+    types: [CellValueType.number, CellValueType.text, CellValueType.boolean, CellValueType.error],
+    compute: (values) => countAny([[values]]),
+  },
+  {
+    name: _t("Count Numbers"),
+    types: [CellValueType.number, CellValueType.text, CellValueType.boolean, CellValueType.error],
+    compute: (values, locale) => countNumbers([[values]], locale),
+  },
+];
+
+export class AggregateStatisticsStore extends SpreadsheetStore {
+  statisticFnResults: StatisticFnResults = {};
+  private isDirty = false;
+
+  constructor(get: Get) {
+    super(get);
+    this.model.selection.observe(this, {
+      handleEvent: this.handleEvent.bind(this),
+    });
+  }
+
+  handle(cmd: Command) {
+    if (
+      (isCoreCommand(cmd) && cmd.type !== "UPDATE_CELL") ||
+      (cmd.type === "UPDATE_CELL" && "content" in cmd)
+    ) {
+      this.isDirty = true;
+    }
+    switch (cmd.type) {
+      case "ACTIVATE_SHEET":
+      case "ACTIVATE_NEXT_SHEET":
+      case "ACTIVATE_PREVIOUS_SHEET":
+      case "EVALUATE_CELLS":
+      case "UNDO":
+      case "REDO":
+        this.isDirty = true;
+    }
+  }
+
+  finalize() {
+    if (this.isDirty) {
+      this.isDirty = false;
+      this.statisticFnResults = this._computeStatisticFnResults();
+    }
+  }
+
+  handleEvent() {
+    if (this.getters.isGridSelectionActive()) {
+      this.statisticFnResults = this._computeStatisticFnResults();
+    }
+  }
+
+  private _computeStatisticFnResults(): StatisticFnResults {
+    const sheetId = this.getters.getActiveSheetId();
+    const cells = new Set<EvaluatedCell>();
+
+    const zones = this.getters.getSelectedZones();
+    for (const zone of zones) {
+      for (const { col, row } of positions(zone)) {
```

Comment:
should we try to replace the `for ... of positions(zone)` by double loops on col/rows to avoid allocating thousands of objects ?

Reply:
I guess we could get rid of positions altogether